components/autofill/src/db/schema.rs

```diff
@@ -90,13 +111,82 @@ fn create(db: &Connection) -> Result<()> {
         )
         .as_str(),
     )?;
-
+    db.execute(
+        &format!("PRAGMA user_version = {version}", version = VERSION),
+        NO_PARAMS,
+    )?;
     Ok(())
 }
 
-#[allow(dead_code)]
 pub fn create_empty_sync_temp_tables(db: &Connection) -> Result<()> {
     log::debug!("Initializing sync temp tables");
     db.execute_batch(CREATE_SYNC_TEMP_TABLES_SQL)?;
     Ok(())
 }
+
+fn upgrade(db: &Connection, from: i64) -> Result<()> {
+    log::debug!("Upgrading schema from {} to {}", from, VERSION);
+    if from == VERSION {
+        return Ok(());
+    }
+    // Places has a cute `migration` helper we can consider using if we get
+    // a few complicated updates, but let's KISS for now.
+    if from == 0 {
+        // This is a bit painful - there are (probably 3) databases out there
+        // that have a schema of 0 but actually exist - ie, we can't assume
+        // a schema of zero implies "new database".
+        // These databases have a `cc_number` but we need them to have a
+        // `cc_number_enc` and `cc_number_last_4`.
+        // This was so very early in the Fenix nightly cycle, and before any
+        // real UI existed to create cards, so we don't bother trying to
+        // migrate them, we just drop the table so it's re-created with the
+        // correct schema.
+        db.execute("DROP TABLE IF EXISTS credit_cards_data", NO_PARAMS)?;
+    }
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::db::test::new_mem_db;
+
+    #[test]
+    fn test_create_schema_twice() {
+        let db = new_mem_db();
+        db.execute_batch(CREATE_SHARED_SCHEMA_SQL)
+            .expect("should allow running main schema creation twice");
+        // sync tables aren't created by default, so do it twice here.
+        db.execute_batch(CREATE_SYNC_TEMP_TABLES_SQL)
+            .expect("should allow running sync temp tables first time");
+        db.execute_batch(CREATE_SYNC_TEMP_TABLES_SQL)
+            .expect("should allow running sync temp tables second time");
+    }
+
+    #[test]
+    fn test_upgrade_version_0() {
+        let db = new_mem_db();
+        // Manually hack things back to where we had to migrate from
+        // version 0.
+        // We don't care what the old table actually has because we drop it
+        // without a migration.
+        db.execute_batch(
```

Comment:
huge nit: the alignment of this query seems off, particularly the quotation marks (I said huge)

Reply:
I know! Clippy insists on that first quote being there, which I don't understand. It does seem fine with the end quote being on the same line, so I guess that's something.